app/src/main/java/com/mobileproto/hireddit/hireddit/reddit/RedditSearcher.java

```diff
@@ -0,0 +1,160 @@
+package com.mobileproto.hireddit.hireddit.reddit;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.android.volley.DefaultRetryPolicy;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.TimeoutError;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.common.io.CharStreams;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Set;
+
+import io.indico.Indico;
+import io.indico.network.IndicoCallback;
+import io.indico.results.IndicoResult;
+import io.indico.utils.IndicoException;
+
+/**
+ * RedditSearcher: Takes in user's spoken string and displays the best comment on the UI
+ * Uses Indico to get key words
+ * Calls GetComment to get Reddit comments with key words
+ * Calls ChooseComment to get the best comment
+ */
+public class RedditSearcher implements Response.Listener<JSONObject>, Response.ErrorListener {
+
+    private String spokenString;
+    private static final String DEBUG_TAG = "RedditSearcher Debug";
+    private static final String ERROR_TAG = "RedditSearcher Error";
+
+    private Context context;
+    private Indico indico;
+    private CommentCallback myCommentCallback;
+    private RequestQueue queue;
+
+    private IndicoCallback<IndicoResult> indicoCallback = new IndicoCallback<IndicoResult>() {
+        @Override
+        public void handle(IndicoResult result) throws IndicoException {
+            getCommentFromKeywords(result.getKeywords().keySet());
+        }
+    };
+
+    public RedditSearcher(CommentCallback myCommentCallback, String spokenString, Context context) {
+        this.myCommentCallback = myCommentCallback;
+        this.spokenString = spokenString;
+        this.context = context;
+        String indicoApiKey = getApi(context);
+        this.indico = Indico.init(context, indicoApiKey, null);
+        this.queue = Volley.newRequestQueue(context);
+    }
+
+    public static String getApi(Context context) {
+        try {
+            AssetManager assetManager = context.getAssets();
+            InputStream apiKey = assetManager.open("indicoapitxt.txt");
+            String apiKeyString = CharStreams.toString(new InputStreamReader(apiKey, "UTF-8"));
+            return apiKeyString;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public void getRedditComment() {
```

Comment:
Well, it indirectly does get a reddit comment, since the indico callback then triggers a reddit search which eventually triggers a callback with the reddit comment in it back to the fragment.